src/core/Merger.h

```diff
@@ -0,0 +1,70 @@
+/*
+ *  Copyright (C) 2018 KeePassXC Team <team@keepassxc.org>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef KEEPASSXC_MERGER_H
+#define KEEPASSXC_MERGER_H
+
+#include "core/Group.h"
+#include <QCoreApplication>
+#include <QPointer>
+
+class Database;
+class Entry;
+
+class Merger
+{
+    Q_DECLARE_TR_FUNCTIONS(Merger)
```

Comment:
Any reason not to make this a QObject?

Reply:
QObject is not needed to work properly for the merger - at least not in the current design as shiort living utility. QObject would add just unnecessary compile time overhead.

Reply:
I don't see the overhead to be problematic compared to the benefits of being able to use the meta object system, signals, slots, and generally more uniformity.

Reply:
I simply don't see any benefits. Without inheriting, the API reveals the only intend the class for and how it should be used. The clutter using the meta system somewhat hides this and suggests uses which the class was not designed for. Currently, adjusting to `QObject` is trivial, but I'd like to have the person changing to inherit to `QObject` reason about the consequences (memory handling, object lifetime, ...). That's why I would prefer `Merger`to be a simple class instead `QObject` based.

Reply:
I am mostly arguing from a point of convention here. This class is not performance-critical, so this bit of overhead rarely matters. We have a few classes which are not derived from QObject and for some it's vital, for most it's quite confusing. I've run into the situation often enough that I wanted to use a QMetaObject function on an instance of a class only to realise that it does not inherit from QObject. Adding QObject later is not as easy as you might think. Sometimes incompatible inheritance of usage patterns have evolved which makes it impossible to let a class inherit from QObject later without refactoring large parts of the code. For a Merger class, I could very well imagine use cases where you would want to use signals and slots later on.

Reply:
To be honest, my experience is, that I don't want to use `QObject` as long I don't have to. Since the change is trivial (for now), so I'll go along and inherit from `QObject`.

Reply:
I won't fight to the death about this one, but I think consistency is a good thing here.
For objects which will exists many times in memory, I would agree with you, though.